app/src/main/java/com/rkm/tasky/network/authentication/implementation/AuthenticationManagerImpl.kt

```diff
@@ -0,0 +1,75 @@
+package com.rkm.tasky.network.authentication.implementation
+
+import com.rkm.tasky.di.IoDispatcher
+import com.rkm.tasky.network.authentication.abstraction.AuthenticationManager
+import com.rkm.tasky.network.model.dto.asAuthInfo
+import com.rkm.tasky.network.repository.implementation.AuthenticationRepositoryImpl
+import com.rkm.tasky.network.util.NetworkError
+import com.rkm.tasky.util.result.EmptyResult
+import com.rkm.tasky.util.result.asEmptyDataResult
+import com.rkm.tasky.util.result.onSuccess
+import com.rkm.tasky.util.storage.implementation.SessionStorageImpl
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class AuthenticationManagerImpl @Inject constructor(
+    private val repository: AuthenticationRepositoryImpl,
+    private val sessionStorage: SessionStorageImpl,
```

Comment:
Here, you depend on an implementation instead of the abstractions that you created. Doing so does defeats the purpose of having the abstraction in the first place by preventing you from swapping the dependency out for another implementation of the abstraction when you want to provide fake implementations to your repository for tests, for example. It also leaks implementation details to this Repo.